lib/pure/asyncmacro.nim

```diff
@@ -440,7 +399,24 @@ macro multisync*(prc: untyped): untyped =
   result.add(asyncSingleProc(asyncPrc))
   result.add(sync)
 
-proc await*[T](x: T) =
-  ## The 'await' keyword is also defined here for technical
-  ## reasons. (Generic symbol lookup prepass.)
-  {.error: "Await only available within .async".}
+# seems simpler to define it here, than to define it in each multisync
+template await*(f: untyped): untyped =
+  when declared(internalRetFuture):
+    error "Await expects a Future"
+  elif not declared(inMultisync):
+    error "Await only available within {.async.}"
+  else:
+    f
+ 
+# based on the yglukhov's patch to chronos: https://github.com/status-im/nim-chronos/pull/47
+template await*[T](f: Future[T]): auto =
+  when declared(internalRetFuture):
+    when not declared(internalTmpFuture):
+      var internalTmpFuture: FutureBase
+    internalTmpFuture = f
+    yield internalTmpFuture
+    if not isNil(internalTmpFuture.error):
+      raise internalTmpFuture.error
+    cast[type(f)](internalTmpFuture).internalRead()
```

Comment:
The above 3 lines change semantics. Just call `read` instead, also use an explicit `return` if possible.

Reply:
ok, i can change to read

Reply:
changed to read